src/main/java/nus/climods/model/ModelManager.java

```diff
@@ -89,6 +91,39 @@ public boolean isModuleOfferedInSemester(String moduleCode, SemestersEnum semest
         return module.get().availableInSemester(semester);
     }
 
+    @Override
+    public Set<LessonTypeEnum> unselectableLessonType(String moduleCode, SemestersEnum semester) {
+        Optional<Module> module = getModule(moduleCode);
+        if (module.isEmpty()) {
+            return null;
+        }
+
+        return module.get().getUnselectableLessonTypeEnums(semester);
+    }
+
+    @Override
+    public boolean isModuleLessonOffered(String moduleCode, SemestersEnum semester,
+                                         LessonTypeEnum lessonType) {
```

Comment:
Created these methods but faced some issues as I kept hitting invocation error when checking if LessonType exists in PickCommand... still in the midst of trying to figure out the bug...